client/src/scene/SceneGameArena.ts

```diff
@@ -130,12 +135,17 @@ export class SceneGameArena extends Phaser.Scene {
     update(time: number, delta: number) {
         this.frameTimeElapsed += delta;
 
+        if (this.trade == null && this.gameState.playerId != null) {
+            this.trade = new TradeUI(this);
+        }
+
         // 12 fps
         if (this.frameTimeElapsed > 1000 / this.FRAMERATE) {
             this.updateBoardFromFrozen(this, this.gameState.otherTetrominoes);
             this.updateUserInput(this);
             this.updateDrawBoard(this.gameState, this);
             this.updateDrawPlayer(this);
+            this.updateFromTradeState(this);
```

Comment:
What was the motivation for calling this function 12 times a second rather than just when the client receives a trade event?

Reply:
Well, the client receives the trade event in gamestate, which doesn't have a direct way to update the scene text in SceneGameArena. So currently, Scene Game Arena updates based on the change of state in gamestate. 

This naturally causes a slight delay in the display which might occasionally be a problem.